spec/services/languages/detection_spec.rb

```diff
@@ -0,0 +1,21 @@
+require "rails_helper"
+
+RSpec.describe Languages::Detection, type: :service do
+  subject { described_class.call(text) }
+
+  context "when the text is clearly identifiable as English" do
+    let(:text) { "This is clearly English text." }
+
+    it "returns en" do
+      expect(subject).to eq(:en)
+    end
+  end
+
+  context "when the text is clearly identifiable as Spanish" do
+    let(:text) { "Esto es claramente un texto en español." }
+
+    it "returns es" do
+      expect(subject).to eq(:es)
+    end
+  end
+end
```

Comment:
These tests do a good job at testing the happy-path. I think the value-add would be to use an `instance_double` to test that `Detection` does "the right thing" (return `nil`) when the probability is low and the result isn't reliable.